src/config.rs

```diff
@@ -50,29 +50,33 @@ pub struct RusticConfig {
 
 impl RusticConfig {
     pub fn merge_profile(&mut self, profile: &str) -> Result<(), FrameworkError> {
-        let mut path = ProjectDirs::from("", "", "rustic").map_or_else(
-            || Path::new(".").to_path_buf(),
-            |path| path.config_dir().to_path_buf(),
-        );
-        if !path.exists() {
-            path = Path::new(".").to_path_buf();
-        };
-        let path = path.join(profile.to_string() + ".toml");
+        let profile_filename = profile.to_string() + ".toml";
+        let paths = get_config_paths(&profile_filename);
 
-        if path.exists() {
+        if let Some(path) = paths.iter().find(|path| path.exists()) {
             // TODO: This should be log::info! - however, the logging config
             // can be stored in the config file and is needed to initialize the logger
             eprintln!("using config {}", path.display());
-            let mut config = Self::load_toml_file(AbsPathBuf::new(&path)?)?;
+            let mut config = Self::load_toml_file(AbsPathBuf::canonicalize(path)?)?;
             // if "use_profile" is defined in config file, merge the referenced profiles first
             for profile in &config.global.use_profile.clone() {
                 config.merge_profile(profile)?;
             }
             self.merge(config);
         } else {
+            let paths_string = paths.iter().fold(String::new(), |acc, path| {
```

Comment:
Why not use `join` from itertools? (It is already in the dependency list)

Reply:
Done. Nice catch! Didn't know it was already included